todo_app/app.py

```diff
@@ -8,162 +8,215 @@
+
+    login_manager.init_app(app)
+
+    def create_items(name):
+        client = pymongo.MongoClient(
+            dbconnect, ssl=True, ssl_cert_reqs='CERT_NONE')
+        database = client["01"]
+        date_now = datetime.now().strftime('%Y-%m-%d')
+
+        post = {"name": name, "idBoard": todoBoard,
+                "last_modified": date_now}
+
+        trello_collection = database["trello_collection"]
+        result = trello_collection.insert_one(post)
+
+    def update_item(id):
+        client = pymongo.MongoClient(
+            dbconnect, ssl=True, ssl_cert_reqs='CERT_NONE')
+        database = client["01"]
+        post = {"_id": ObjectId(id)}
+        trello_collection = database["trello_collection"]
+        date_now = datetime.now().strftime('%Y-%m-%d')
+        result = trello_collection.update_one(
+            post, {"$set": {"idBoard": doneBoard,
+                            "last_modified": date_now}}
+        )
+
+    def return_todo(id):
+        client = pymongo.MongoClient(
+            dbconnect, ssl=True, ssl_cert_reqs='CERT_NONE')
+        database = client["01"]
+        post = {"_id": ObjectId(id)}
+        trello_collection = database["trello_collection"]
+        date_now = datetime.now().strftime('%Y-%m-%d')
+        result = trello_collection.update_one(
+            post, {"$set": {"idBoard": todoBoard,
+                            "last_modified": date_now}}
+        )
+
+    @app.route('/')
+    @login_required
+    def index():
+        connectDb()
+        client = pymongo.MongoClient(
+            dbconnect, ssl=True, ssl_cert_reqs='CERT_NONE')
+        database = client["01"]
+        trello_collection = database["trello_collection"]
+        todo = trello_collection.find(
+            {'idBoard': todoBoard}, {"name": 1})
+        doing = trello_collection.find(
+            {'idBoard': doingBoard}, {"name": 1})
+        done = trello_collection.find(
+            {'idBoard': doneBoard}, {"name": 1})
+
+        my_items = []
+        doing_objects = []
+        done_objects = []
+
+        for todo in trello_collection.find({'idBoard': todoBoard}):
+            my_items.append(ToDo.from_mongo_db_entry(todo))
+        for doing in trello_collection.find({'idBoard': doingBoard}):
+            doing_objects.append(ToDo.from_mongo_db_entry(doing))
+        for done in trello_collection.find({'idBoard': doneBoard}):
+            done_objects.append(ToDo.from_mongo_db_entry(done))
+        view_model = ViewModel(my_items, doing_objects, done_objects)
+        view_model.show_all_done_items()
+        user = current_user.role
+
+        return render_template("index.html", my_items=my_items, doing_objects=doing_objects, done_objects=done_objects, view_model=view_model, user=user)
+
+    @app.route('/create', methods=['POST'])
+    @login_required
+    def create():
+        print(current_user.id)
+        title = request.form.get('title')
+        create_items(title)
+
+        return redirect(url_for('index'))
```

Comment:
You haven't actually secured your application yet. Currently, anyone with a GitHub account can log in and then send a POST request to this /create endpoint.

You're correctly hiding the form, but that doesn't protect you - anyone could manually send a POST request (with a tool like Postman for example)

Reply:
Updated, thank you